plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -571,6 +573,23 @@ open class DefaultPageCreator(
         }
     }
 
+    /**
+     * Divergent elements, such as extensions for the same receiver, can have identical signatures
+     * if they are declared in different places. If such elements are shown on the same page together,
+     * they need to be rendered deterministically to have reproducible builds.
+     *
+     * For example, you can have three identical extensions, if they are declared as:
+     * 1) top-level in package A
+     * 2) top-level in package B
+     * 3) inside a companion object in package A/B
+     *
+     * @see divergentBlock
+     */
+    private fun sortDivergentElementsDeterministically(elements: List<Documentable>): List<Documentable> =
```

Comment:
Minor comment: it sorts deterministically only by signature.

Reply:
Not sure if you're proposing something

Also not sure what you mean by `only by signature` - are containing packages and classes part of the signature? :thinking:

Reply:
I mean if some elements have the same signature then the sort is not deterministic anymore. My proposal is to rename it to e.g. `sortDivergentElementsBySignature`.

Reply:
> if some elements have the same DRI

I thought DRI is supposed to be a unique identifier :thinking: 

Can you give an example of such Kotlin code that has 2 functions/properties, it compiles, and both declarations have the same DRI?

Reply:
Sorry for my bad explanation)
I confused myself.
- This sort is deterministic since  `sortedWith` is a stable one so something like `sortDivergentElements` can be better.
 - But a comparator can be deterministic as well, but I am not sure about DRI and other fields.
 As for me, it is too formal to have `deterministic` postfix for the comparator. 
 The naming is up to you. 
  
> I thought DRI is supposed to be a unique identifier thinking

Only in compilable projects.  Some users call Dokka with arbitrary source sets.

Reply:
> As for me, it is too formal to have deterministic postfix for the comparator.

The comparator doesn't have the `deterministic` part, it's called `divergentDocumentableComparator`.

But it is indeed very formal to have `deterministic` in the name of the function - that's the whole purpose of it. It's not just some sorting, it solves the problem of deterministic rendering, which, if someone starts refactoring things or fixing nearby bugs, must be respected

Reply:
> The comparator doesn't have the deterministic part, it's called divergentDocumentableComparator.

Sorry, it would be too formal if it even had `deterministic`.

> It's not just some sorting

Without context it seems about sorting.